### PR TITLE
Better argument processing. Resolves #15, #19, #27.

### LINKED ISSUE #15: Automatically stop profiling after a specified duration
Reintroduce the "duration:N" parameter support in the profiler by having it detect that the specified number of seconds has elapsed and stop automatically. This requires stopping the profiler from a separate thread, or after returning from the signal handler, because it is not safe to stop the profiler from the signal handler that detects the timeout.

Perhaps we should consider leaving this up to the user, and removing the "duration" parameter altogether? Alternatively, the helper script could accept a duration parameter and do this automatically:

```
$ ./profiler.sh -p 8983 -a start -d 10      # duration: 10 seconds, script stops automatically
```

### PR DESCRIPTION
**Pull request by apangin**
Here is a number of improvements both to parsing of Agent arguments and to profiler.sh command-line interface.

1. Moved arguments processing to the separate `arguments.cpp` file.
2. Removed the timeout check from the agent code (it did not work anyway). Now it is responsibility of the wrapper script to stop the profiling after the specified period of time.
3. Added `-d` (duration) option to profiler.sh. It allows to execute start-wait-stop sequence with just one command.
4. Allowed to dump any combination of profiling data (summary / flat profile / call stacks / flamegraph) also with just one command.
5. Filename option now works not only for the flamegraph format, but for any combination of output formats above.
6. Added --status option to print whether profiler is active and for how long.

### CODE REVIEW

**Comment by apangin** [comment]
Agree. I didn't feel comfortable with -- options either. Replaced them with actions. If action is omitted, timed profiling will take place.

**Comment by apangin** [comment]
One more addition:

- Any profiler output can now be redirected to a file, including messages printed by "start", "stop" and "status" commands.
- If no `-f` option is given, `profiler.sh` makes agent dump to a temporary file, which is then printed to a caller console. It is no longer necessary to switch between terminals; profiling output is printed to the script's console by default.

File: profiler.sh [hunk 1]
```diff
@@ -74,22 +86,30 @@ done
 
 [[ "$PID" == "" ]] && usage
 
+# if no -f argument is given, use temporary file to transfer output to caller terminal
+if [[ $USE_TMP ]]; then
+    FILE="/tmp/async-profiler-${PID}.tmp";
```
Comment (goldshtn) [inline profiler.sh]: Wouldn't mktemp be safer?

**Comment by goldshtn** [comment]
LGTM except the temp file name comment. I've written a quick blog post about async-profiler which I plan to publish tomorrow, just FYI.

File: profiler.sh (hunk 1 above)
Comment (apangin) [inline profiler.sh]: All right; changed to mktemp.

**Comment by apangin** [comment]
Greate, thank you!
Please note that I've moved the repository to jvm-profiling-tools as @nitsanw suggested, so make sure you've updated the links in your post.

**Pull request merged**